### PR TITLE
Fixes bug 1188956 - Added a script to migrate data from PostgreSQL to…

### PR DESCRIPTION
**Pull request by AdrianGaudebert**
… any database.

@peterbe r?
cc @rhelmer 

``fakedata`` was inserting very diverse data into the ``reports`` table but always the same data in the ``processed_crashes`` table, so I had to rework the script a bit to make it insert the same diverse data in both tables.

### CODE REVIEW

File: socorro/external/postgresql/crash_migration_app.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        except Exception:
+            self.config.logger.warning(
+                'error loading crash %s',
+                crash_id,
+                exc_info=True
+            )
+            return
+
+        if 'uuid' not in raw_crash:
+            raw_crash.uuid = crash_id
```
Comment (peterbe) [inline socorro/external/postgresql/crash_migration_app.py]: Strange that that would ever happen. Isn't this based on `select uuid from raw_crashes`? And what is `raw_crash` here? A dict? A DotDict?

File: socorro/external/postgresql/fakedata.py [hunk 2]
```diff
@@ -392,6 +392,130 @@ def daily_builds(self, fragment, channel, days=None):
...
+                                   hangid,
+                                   process_type,
+                                   channel_name,
+                                   product_guid,
+                                   exploitability,
+                                   channel_name]
+
+                            yield row
+                            count += 1
+
```
Comment (peterbe) [inline socorro/external/postgresql/fakedata.py]: Is this thing different from what it was when it belonged to the `Products` class?

**Comment by peterbe** [comment]
I know it sounds like a lame excuse but I can't review this by testing because I can't even get setupdb_app.py to work:
```
(socorro):~/dev/MOZILLA/SOCORRO/socorro ((e62b45d...) %)$ python socorro/external/postgresql/setupdb_app.py --dropdb --database_name=socorro_integration_test --database_username=peterbe --database_password=test123 --database_superusername=peterbe --database_superuserpassword=test123 --fakedata --fakedata_days=28
Traceback (most recent call last):
  File "socorro/external/postgresql/setupdb_app.py", line 452, in <module>
    sys.exit(main(SocorroDBApp))
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/app/socorro_app.py", line 493, in main
    values_source_list=values_source_list
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/app/socorro_app.py", line 178, in run
    values_source_list=values_source_list
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/app/socorro_app.py", line 276, in _do_run
    return_code = fix_exit_code(app_to_run.main())
  File "socorro/external/postgresql/setupdb_app.py", line 387, in main
    on_heroku=self.config.on_heroku
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/external/postgresql/postgresqlalchemymanager.py", line 31, in __init__
    self.conn = self.engine.connect().execution_options(
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1778, in connect
    return self._connection_cls(self, **kwargs)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 60, in __init__
    self.__connection = connection or engine.raw_connection()
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1847, in raw_connection
    return self.pool.unique_connection()
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/pool.py", line 280, in unique_connection
    return _ConnectionFairy._checkout(self)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/pool.py", line 644, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/pool.py", line 440, in checkout
    rec = pool._do_get()
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/pool.py", line 963, in _do_get
    return self._create_connection()
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/pool.py", line 285, in _create_connection
    return _ConnectionRecord(self)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/pool.py", line 411, in __init__
    self.connection = self.__connect()
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/pool.py", line 538, in __connect
    connection = self.__pool._creator()
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/engine/strategies.py", line 96, in connect
    connection_invalidated=invalidated
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 199, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/engine/strategies.py", line 90, in connect
    return dialect.connect(*cargs, **cparams)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 377, in connect
    return self.dbapi.connect(*cargs, **cparams)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/psycopg2/__init__.py", line 164, in connect
    conn = _connect(dsn, connection_factory=connection_factory, async=async)
sqlalchemy.exc.OperationalError: (OperationalError) FATAL:  database "socorro_integration_test" does not exist
 None None
```

Creating the database (`createdb socorro_integration_test`) isn't helping. Same error. 

Any idea?

I guess my chances of debugging it are as good as yours but perhaps because you've worked with it more recently you might have a bit headstart.

File: socorro/external/postgresql/crash_migration_app.py (hunk 1 above)
Comment (AdrianGaudebert) [inline socorro/external/postgresql/crash_migration_app.py]: Oh, this is just copy-pasta leftover. It indeed makes no sense here!

File: socorro/external/postgresql/fakedata.py (hunk 2 above)
Comment (AdrianGaudebert) [inline socorro/external/postgresql/fakedata.py]: Nope, I didn't change a line of it!

**Comment by AdrianGaudebert** [comment]
There's a "new" parameter to setupdb that you need to set: 

``socorro setupdb --fakedata --fakedata_days=3 --createdb``

And if you want to delete the previous database (which you will need to do if you have one, you can't just add data on top of an existing database with fakedata): 

``socorro setupdb --fakedata --fakedata_days=3 --createdb --dropdb --force``

**Comment by peterbe** [comment]
Perhaps it's because I forgot to start my local ES server, but it ends quite strangely:
```
(socorro):~/dev/MOZILLA/SOCORRO/socorro ((e62b45d...) %)$ python socorro/external/postgresql/setupdb_app.py --dropdb --database_name=socorro_integration_test --database_username=peterbe --database_password=test123 --database_superusername=peterbe --database_superuserpassword=test123 --fakedata --fakedata_days=7 --createdb
(socorro):~/dev/MOZILLA/SOCORRO/socorro ((e62b45d...) %)$ python socorro/external/postgresql/crash_migration_app.py
2015-08-04 09:40:42,285 INFO - crash_migration -  - MainThread - app_name: crash_migration
2015-08-04 09:40:42,285 INFO - crash_migration -  - MainThread - app_version: 1.0
2015-08-04 09:40:42,285 INFO - crash_migration -  - MainThread - current configuration:
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - application: <class 'socorro.external.postgresql.crash_migration_app.CrashMigrationApp'>
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.backoff_delays: [10, 30, 60, 120, 300]
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.crashstorage_class: <class 'socorro.external.es.crashstorage.ESCrashStorageNoStackwalkerOutput'>
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.elasticsearch.elasticsearch_class: <class 'socorro.external.es.connection_context.ConnectionContext'>
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.elasticsearch.elasticsearch_connection_wrapper_class: <class 'socorro.external.es.connection_context.Connection'>
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.elasticsearch.elasticsearch_default_index: socorro
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.elasticsearch.elasticsearch_doctype: crash_reports
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.elasticsearch.elasticsearch_index: socorro%Y%W
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.elasticsearch.elasticsearch_timeout: 30
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.elasticsearch.elasticsearch_urls: ['localhost:9200']
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.es_redactor.forbidden_keys: json_dump, upload_file_minidump_flash1.json_dump, upload_file_minidump_flash2.json_dump, upload_file_minidump_browser.json_dump
2015-08-04 09:40:42,286 INFO - crash_migration -  - MainThread - destination.es_redactor.redactor_class: <class 'socorro.external.crashstorage_base.Redactor'>
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - destination.forbidden_keys: json_dump, upload_file_minidump_flash1.json_dump, upload_file_minidump_flash2.json_dump, upload_file_minidump_browser.json_dump
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - destination.redactor_class: <class 'socorro.external.crashstorage_base.Redactor'>
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - destination.transaction_executor_class: <class 'socorro.database.transaction_executor.TransactionExecutorWithLimitedBackoff'>
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - destination.wait_log_interval: 10
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - logger: <socorro.app.socorro_app.LoggerWrapper object at 0x10af27050>
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - logging.stderr_error_logging_level: 10
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - logging.stderr_line_format_string: {asctime} {levelname} - {app_name} - {message}
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - logging.syslog_error_logging_level: 40
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - logging.syslog_facility_string: user
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - logging.syslog_host: localhost
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - logging.syslog_line_format_string: {app_name} (pid {process}): {asctime} {levelname} - {threadName} - {message}
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - logging.syslog_port: 514
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - new_crash_source.backoff_delays: [10, 30, 60, 120, 300]
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - new_crash_source.crash_id_query: select uuid from reports_20141020 where uuid like '%142022' and date_processed > '2014-10-23'
2015-08-04 09:40:42,287 INFO - crash_migration -  - MainThread - new_crash_source.crashstorage_class: <class 'socorro.external.postgresql.crashstorage.PostgreSQLCrashStorage'>
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.database_class: <class 'socorro.external.postgresql.connection_context.ConnectionContext'>
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.database_hostname: localhost
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.database_name: breakpad
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.database_password: *********
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.database_port: 5432
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.database_username: breakpad_rw
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.forbidden_keys: json_dump, upload_file_minidump_flash1.json_dump, upload_file_minidump_flash2.json_dump, upload_file_minidump_browser.json_dump
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.new_crash_source_class: <class 'socorro.processor.timemachine.PGQueryNewCrashSource'>
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.redactor_class: <class 'socorro.external.crashstorage_base.Redactor'>
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.transaction_executor_class: <class 'socorro.database.transaction_executor.TransactionExecutorWithInfiniteBackoff'>
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - new_crash_source.wait_log_interval: 10
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - producer_consumer.idle_delay: 7
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - producer_consumer.maximum_queue_size: 8
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - producer_consumer.number_of_threads: 4
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - producer_consumer.producer_consumer_class: <class 'socorro.lib.threaded_task_manager.ThreadedTaskManager'>
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - producer_consumer.quit_on_empty_queue: False
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - resource.elasticsearch.elasticsearch_class: <class 'socorro.external.es.connection_context.ConnectionContext'>
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - resource.elasticsearch.elasticsearch_connection_wrapper_class: <class 'socorro.external.es.connection_context.Connection'>
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - resource.elasticsearch.elasticsearch_default_index: socorro
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - resource.elasticsearch.elasticsearch_doctype: crash_reports
2015-08-04 09:40:42,288 INFO - crash_migration -  - MainThread - resource.elasticsearch.elasticsearch_index: socorro%Y%W
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.elasticsearch.elasticsearch_timeout: 30
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.elasticsearch.elasticsearch_urls: ['localhost:9200']
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.logging.stderr_error_logging_level: 10
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.logging.stderr_line_format_string: {asctime} {levelname} - {app_name} - {message}
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.logging.syslog_error_logging_level: 40
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.logging.syslog_facility_string: user
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.logging.syslog_host: localhost
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.logging.syslog_line_format_string: {app_name} (pid {process}): {asctime} {levelname} - {threadName} - {message}
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.logging.syslog_port: 514
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.postgresql.backoff_delays: [10, 30, 60, 120, 300]
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.postgresql.crashstorage_class: <class 'socorro.external.postgresql.crashstorage.PostgreSQLCrashStorage'>
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.postgresql.database_class: <class 'socorro.external.postgresql.connection_context.ConnectionContext'>
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.postgresql.database_hostname: localhost
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.postgresql.database_name: breakpad
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.postgresql.database_port: 5432
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.postgresql.transaction_executor_class: <class 'socorro.database.transaction_executor.TransactionExecutorWithInfiniteBackoff'>
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.postgresql.wait_log_interval: 10
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.redactor.forbidden_keys: json_dump, upload_file_minidump_flash1.json_dump, upload_file_minidump_flash2.json_dump, upload_file_minidump_browser.json_dump
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - resource.redactor.redactor_class: <class 'socorro.external.crashstorage_base.Redactor'>
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - secrets.postgresql.database_password: *********
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - secrets.postgresql.database_username: breakpad_rw
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - source.backoff_delays: [10, 30, 60, 120, 300]
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - source.crashstorage_class: <class 'socorro.external.postgresql.crash_migration_app.DumbPostgreSQLCrashStorage'>
2015-08-04 09:40:42,289 INFO - crash_migration -  - MainThread - source.database_class: <class 'socorro.external.postgresql.connection_context.ConnectionContext'>
2015-08-04 09:40:42,290 INFO - crash_migration -  - MainThread - source.database_hostname: localhost
2015-08-04 09:40:42,290 INFO - crash_migration -  - MainThread - source.database_name: breakpad
2015-08-04 09:40:42,290 INFO - crash_migration -  - MainThread - source.database_password: *********
2015-08-04 09:40:42,290 INFO - crash_migration -  - MainThread - source.database_port: 5432
2015-08-04 09:40:42,290 INFO - crash_migration -  - MainThread - source.database_username: breakpad_rw
2015-08-04 09:40:42,290 INFO - crash_migration -  - MainThread - source.forbidden_keys: json_dump, upload_file_minidump_flash1.json_dump, upload_file_minidump_flash2.json_dump, upload_file_minidump_browser.json_dump
2015-08-04 09:40:42,290 INFO - crash_migration -  - MainThread - source.redactor_class: <class 'socorro.external.crashstorage_base.Redactor'>
2015-08-04 09:40:42,290 INFO - crash_migration -  - MainThread - source.transaction_executor_class: <class 'socorro.database.transaction_executor.TransactionExecutorWithInfiniteBackoff'>
2015-08-04 09:40:42,290 INFO - crash_migration -  - MainThread - source.wait_log_interval: 10
2015-08-04 09:40:42,290 INFO - crash_migration -  - MainThread - installing signal handers
2015-08-04 09:40:42,290 DEBUG - crash_migration -  - MainThread - TaskManager finished init
2015-08-04 09:40:42,290 WARNING - crash_migration -  - MainThread - beware, this crashstorage class is destructive to the processed crash - if you're using a polycrashstore you may find the modified processed crash saved to the other crashstores
2015-08-04 09:40:42,290 DEBUG - crash_migration -  - MainThread - start
2015-08-04 09:40:42,291 DEBUG - crash_migration -  - QueuingThread - _queuing_thread_func start
2015-08-04 09:40:42,291 DEBUG - crash_migration -  - MainThread - waiting to join queuingThread
2015-08-04 09:40:42,321 DEBUG - crash_migration -  - QueuingThread - the loop didn't actually loop
2015-08-04 09:40:42,321 DEBUG - crash_migration -  - QueuingThread - we're quitting queuingThread
2015-08-04 09:40:42,321 DEBUG - crash_migration -  - QueuingThread - waiting for standard worker threads to stop
2015-08-04 09:40:42,321 DEBUG - crash_migration -  - QueuingThread - all worker threads stopped
```

**Comment by peterbe** [comment]
Oh actually it quits the same strange way even after I've started ES on localhost:9200

**Comment by AdrianGaudebert** [comment]
@peterbe you create and populate the ``socorro_integration_test`` database, and then use ``breakpad`` in the code. :)

**Comment by peterbe** [comment]
One of many errors:
```
2015-08-04 09:50:23,854 ERROR - crash_migration -  - Thread-2 - Error in processing a job
Traceback (most recent call last):
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/lib/threaded_task_manager.py", line 342, in run
    function(*args, **kwargs)  # execute the task
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/app/fetch_transform_save_app.py", line 141, in transform
    self._transform(crash_id)
  File "socorro/external/postgresql/crash_migration_app.py", line 94, in _transform
    crash_id
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/external/es/crashstorage.py", line 215, in save_raw_and_processed
    crash_id
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/external/es/crashstorage.py", line 94, in save_raw_and_processed
    crash_document=crash_document
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/database/transaction_executor.py", line 114, in __call__
    result = function(connection, *args, **kwargs)
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/external/es/crashstorage.py", line 121, in _submit_crash_to_elasticsearch
    index_creator.create_socorro_index(es_index)
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/external/es/index_creator.py", line 59, in create_socorro_index
    es_settings = SuperSearchFields(config=self.config).get_mapping()
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/external/es/super_search_fields.py", line 302, in get_mapping
    all_fields = self.get_fields()
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/external/es/super_search_fields.py", line 35, in get_fields
    doc_type='supersearch_fields',
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/elasticsearch/client/utils.py", line 68, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/elasticsearch/client/__init__.py", line 622, in count
    params=params, body=body)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/elasticsearch/transport.py", line 284, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/elasticsearch/connection/http_requests.py", line 54, in perform_request
    self._raise_error(response.status_code, raw_data)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/elasticsearch/connection/base.py", line 97, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
NotFoundError: TransportError(404, u'IndexMissingException[[socorro] missing]')
```

Do I need to run some sort of setupES_app.py first?

**Comment by peterbe** [comment]
I generated 7 days of fakedata on PG and that didn't take long. A couple of tens of seconds.
However, running crash_migration_app.py took hours! I started it before lunch and it's been running for a long time even after. 

Is this really the best we can do? It makes it pretty sadface to use. There has got to be some other bulk operation we can use. If not, perhaps rewrite some loop so that it does 20 crashes at a time instead of 1 at a time.

**Comment by peterbe** [comment]
Well. At least it works. 
screenshot 2015-08-04 14 31 25

So it's a reluctant r+
I hope I never have to do this again :)

**Comment by AdrianGaudebert** [comment]
@peterbe https://bugzilla.mozilla.org/show_bug.cgi?id=1179456 added a bulk indexing crash storage class, but it disappeared from master for some reason. As soon as it's back in, let's indeed make use of that one. It should be a big speed win.

**Comment by AdrianGaudebert** [comment]
I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1191240 as a follow-up.

**Pull request merged**